src/Chunk.ts

```diff
@@ -684,24 +681,32 @@ export default class Chunk {
 					if (visitDep(subDep)) return true;
 				}
 			}
-			return visitor(dep) === true;
+			return isConditionSatisfied(dep) === true;
 		}
 		return visitDep(this);
 	}
 
+	visitDependencies(handleDependency: (dependency: Chunk | ExternalModule) => void) {
+		const toBeVisited: (Chunk | ExternalModule)[] = [this];
+		const visited: Set<Chunk | ExternalModule> = new Set();
+		for (const current of toBeVisited) {
+			handleDependency(current);
+			if (current instanceof ExternalModule) continue;
+			for (const dependency of current.dependencies.concat(current.dynamicDependencies)) {
+				if (!visited.has(dependency)) {
+					visited.add(dependency);
+					toBeVisited.push(dependency);
+				}
+			}
+		}
+	}
+
 	private computeContentHashWithDependencies(addons: Addons, options: OutputOptions): string {
 		const hash = sha256();
 
-		// own rendered source, except for finalizer wrapping
-		hash.update(this.getRenderedHash());
```

Comment:
We do need the rendered hash of the source itself here. How else would changing the source without changing any other dependency information change the hash?

Reply:
The rendered hash is added in `visitDependencies`. I made a note to add a new test category at some point to prove this is actually the case.

Reply:
It might be somewhat confusing, though, that visitDependencies always visits the base chunk first. Maybe the name should be improved.